File: Ruby/civitas/pkg/civitas-0.0.1/lib/casilla.rb
#encoding: utf­8

require_relative 'titulo_propiedad'
require_relative 'mazo_sorpresa'
require_relative 'sorpresa'
require_relative 'operaciones_juego'

module Civitas
  class Casilla
    attr_accessor :nombre
    
    def initialize(_nombre)
      @nombre=_nombre
    end
        
   def informe(iactual, todos)#METODO DE INSTANCIA
     Diario.instance.ocurre_evento("El jugador" + todos.at(iactual).to_sj() + " ha caido en la casilla" + self.to_sc())
   end
   
   def jugador_correcto(iactual,todos)
      if(iactual >=0 && todos.size() > iactual)
        return true
      else
        return false
      end
    end

    def recibe_jugador(iactual,todos)
      self.informe(iactual, todos)
    end
   
    def to_sc()
      return "El nombre de la casilla es " + @nombre.to_s()
    end
    

  end
end

File: Java/civitas/src/civitas/Casilla.java

package civitas;

import java.util.ArrayList;

public class Casilla {
    private String nombre;//ATRIBUTO DE INSTANCIA
    
    Casilla(String _nombre){
        nombre=_nombre;
    }
    
    public String getNombre(){
        return nombre;
    }

    public void informe(int iActual, ArrayList<Jugador> todos){
        Diario.getInstance().ocurreEvento("El jugador " + todos.get(iActual).getNombre() + " ha caido en la casilla" +this.toString());
    }
    
    public boolean jugadorCorrecto(int iActual, ArrayList<Jugador> todos){
        if(iActual>=0 && iActual<todos.size())
            return true;
        else
            return false;
    }

    void recibeJugador(int iActual, ArrayList<Jugador> todos){
        this.informe(iActual, todos);
    }
    
    @Override
    public String toString(){
        return "\n"+nombre;
    }  
}
